src/seedu/addressbook/Main.java

```diff
@@ -102,13 +104,19 @@ private void recordResult(CommandResult result) {
      *
      * @param command user command
      * @return result of the command
+     * @throws StorageOperationException 
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            storage.checkStorageFile();
```

Comment:
In this context `storagecheckStorageFile()` is on a lower abstraction level than the rest of the method.
Since `StorageFile` handles file related methods, `storage.checkStorageFile()`  should be called in `storage.save()` in this implementation.